packages/meta-data-tools/src/types.ts

```diff
@@ -1,16 +1,35 @@
-import * as React from 'react';
+import { ComponentType } from 'react';
 
-export interface MetaDataToolsDefinition {
-  metaData: Map<typeof React.Component | React.StatelessComponent<any>, MetaDataDefinition>;
-  describe: (comp: typeof React.Component | React.StatelessComponent<any>) => MetaDataDefinition;
+export interface IRegistry<AssetMap = any> {
+  metadata: IMetadata;
+  describeComponent:<Props> (comp: ComponentType<Props>) => IComponentMetadata<Props>;
+  describeAsset:<AssetType extends keyof AssetMap, Asset extends AssetMap[AssetType]> (asset: Asset, type: AssetType, name: string, description?: string) => IAssetMetadata<AssetType>;
+  describeTheme: (style: any) => IStyleMetadata;
```

Comment:
I think 'theme' is something more general than a style, it provides styles to multiple components. It can also provide colors, typography, icons, etc.